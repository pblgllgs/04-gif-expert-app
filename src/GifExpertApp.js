import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

export const GifExpertApp = ({defaulCategories = []}) => {
    // const [categories, setCategories] = useState(["Hunter X"]);
    const [categories, setCategories] = useState(defaulCategories);
    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories} />
            <hr />
            <ol>
                {categories.map((category) => {
                    return <GifGrid category={category} key={category} />;
                })}
            </ol>
        </>
    );
};

export default GifExpertApp;
