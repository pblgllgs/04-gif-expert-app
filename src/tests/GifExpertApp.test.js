import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Pruebas en el GifExpertApp", () => {
    const wrapper = shallow(<GifExpertApp />);
    test("Debe mostrarse correctamente <GifExpertApp />", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['one punch','gragon ball']
        const wrapper = shallow(<GifExpertApp defaulCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    })
});
