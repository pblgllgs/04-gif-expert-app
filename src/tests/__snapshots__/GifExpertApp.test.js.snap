// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="gragon ball"
      key="gragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el GifExpertApp Debe mostrarse correctamente <GifExpertApp /> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
